sys/include/fido2/ctap/ctap_crypto.h

```diff
@@ -0,0 +1,209 @@
+ * @param[in] in_size          size of data to be encrypted
+ * @param[in] key              symmetric key to use for encryption
+ * @param[in] key_size         size of key
+ *
+ * @return CTAP status code
+ */
+int fido2_ctap_crypto_aes_enc(uint8_t * out, size_t *out_size, uint8_t * in,
+                              size_t in_size, const uint8_t * key, size_t key_size);
+
+/**
+ * @brief Decrypt data using AES-256-CBC
+ *
+ * @param[in] out              decrypted data
+ * @param[in] out_size         size of decrypted data
+ * @param[in] in               encrypted data
+ * @param[in] in_size          size of encrypted data
+ * @param[in] key              symmetric key to use for decryption
+ * @param[in] key_size         size of key
+ *
+ * @return CTAP status code
+ */
+int fido2_ctap_crypto_aes_dec(uint8_t * out, size_t *out_size, uint8_t * in,
+                              size_t in_size, const uint8_t * key, size_t key_size);
+
+/**
+ * @brief Encrypt data using AES-128-CCM
+ *
+ * @param[in] out              encrypted data
+ * @param[in] out_size         size of encrypted data
+ * @param[in] in               data to be encrypted
+ * @param[in] in_size          size of data to be encrypted buffer
+ * @param[in] auth_data        additional data to authenticate in MAC
+ * @param[in] auth_data_size   size of auth_data
+ * @param[in] mac_size         size of appended MAC
+ * @param[in] length_encoding  max supported length of plaintext
+ * @param[in] nonce            nonce for ctr mode encryption
+ * @param[in] nonce_size       size of nonce
+ * @param[in] key              symmetric key to use for encryption
+ * @param[in] key_size         size of key
+ *
+ * @return CTAP status code
+ */
+int fido2_ctap_crypto_aes_ccm_enc(uint8_t *out, size_t out_size,
+                                  const uint8_t *in, size_t in_size,
+                                  uint8_t *auth_data, size_t auth_data_size,
+                                  uint8_t mac_size, uint8_t length_encoding,
+                                  const uint8_t *nonce, size_t nonce_size,
+                                  const uint8_t *key, size_t key_size);
+
+/**
+ * @brief Encrypt data using AES-128-CCM
+ *
+ * @param[in] out              encrypted data
+ * @param[in] out_size         size of encrypted data
+ * @param[in] in               data to be encrypted
+ * @param[in] in_size          size of data to be encrypted buffer
+ * @param[in] auth_data        additional data to authenticate in MAC
+ * @param[in] auth_data_size   size of auth_data
+ * @param[in] mac_size         size of appended MAC
+ * @param[in] length_encoding  max supported length of plaintext
+ * @param[in] nonce            nonce for ctr mode encryption
+ * @param[in] nonce_size       size of nonce
+ * @param[in] key              symmetric key to use for encryption
+ * @param[in] key_size         size of key
+ *
+ * @return CTAP status code
+ */
+int fido2_ctap_crypto_aes_ccm_dec(uint8_t *out, size_t out_size,
```

Comment:
As commented above, I did not quite see the reason for an other abstraction layer on top of the ciphers abstraction. The amount of parameters of this function indicates that something is odd.